ProfitApp/src/main/java/bsu/fpmi/profit/ProfitCRUD.java

```diff
@@ -0,0 +1,424 @@
+                ad.getHashTags() != null && ad.getHashTags().size() >= 1 && ad.getHashTags().size() <= 7 &&
+                ad.getDiscount() != null && ad.getDiscount().length() >= 1 &&
+                ad.getReviews() != null;
+    }
+
+    private PreparedStatement buildAddPreparedStatement(int offerId, int userId, Ad offer) throws SQLException {
+        String sql = "INSERT INTO OFFER(OFFER_ID, USER_ID, OFFER_NAME, DESCRIPTION, VENDOR_LINK, VALID_UNTIL, DISCOUNT, CREATED_AT, PHOTO_LINK) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, offerId);
+        preparedStatement.setInt(2, userId);
+        preparedStatement.setString(3, offer.getLabel());
+        preparedStatement.setString(4, offer.getDescription());
+        preparedStatement.setString(5, offer.getLink());
+        preparedStatement.setDate(6, new java.sql.Date(offer.getValidUntil().getTime()));
+        preparedStatement.setInt(7, Integer.parseInt(offer.getDiscount()));
+        preparedStatement.setTimestamp(8, new java.sql.Timestamp(offer.getCreatedAt().getTime()));
+        preparedStatement.setString(9, offer.getPhotoLink());
+        return preparedStatement;
+    }
+
+    private void addHashtags(Ad offer, int offerId) throws SQLException {
+        String sql = "INSERT INTO HASHTAGS(OFFER_ID, TAG) VALUES (?, ?);";
+        for (String tag : offer.getHashTags()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, offerId);
+            preparedStatement.setString(2, tag);
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+        }
+    }
+
+    public boolean addOffer(Ad offer){
+        if(validateOffer(offer)) {
+            try {
+
+                String sql = "select user_id from user where username = ?";
+                PreparedStatement preparedStatement = connection.prepareStatement(sql);
+                preparedStatement.setString(1, offer.getVendor());
+                ResultSet resultSet = preparedStatement.executeQuery();
+
+
+                int userId;
+                if (resultSet.next()) {
+                    userId = resultSet.getInt("user_id");
+                    preparedStatement.close();
+                    int offerId = (int) (offer.getCreatedAt().getTime() % Integer.MAX_VALUE);
+
+                    preparedStatement = buildAddPreparedStatement(offerId, userId, offer);
+
+                    preparedStatement.executeUpdate();
+                    preparedStatement.close();
+
+                    addHashtags(offer, userId);
+
+                    return true;
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return false;
+    }
+
+    private String getEditSqlString(Ad ad){
```

Comment:
This method does not use any of the class's inner states (such as connection, etc.) Therefore it is sensible to make it `private static`